File: TTT VSC WPF/TicTacWPF/view/Controller_Spiel.cs
using model;
namespace TicTacWPF
{
    public class Controller_Spiel
    {
        private ISpielfeld Spielfeld;
        private ISpieler[] Spieler;
        private ISpielregeln Spielregeln;
        private int Spieleranzahl = 2;
        private int AnDerReihe = 0;
        private int max_id, row, col, win;
        private int generate_Stein_ID = 0;
        private MainWindow MyWindow = null;
        public Controller_Spiel():this(null, 3, 3,3)
        {
        }
        public Controller_Spiel(MainWindow mwd) : this(mwd, 3,3,3)
        {            
        }

        public Controller_Spiel(MainWindow mwd, int prow, int pcol, int pwin){

            row = prow;
            col = pcol;
            win = pwin;
            MyWindow = mwd;
            max_id = Spieleranzahl - 1;
            Spieler Spieler1 = new Spieler("Alice Little", Farbe.Blau, 0);
            Spieler Spieler2 = new Spieler("Cassandra Cain", Farbe.Weiss, 1);
            Spieler = new Spieler[] { Spieler1, Spieler2 };
            Spieleranzahl = 2;
            AnDerReihe = 0;
            generate_Stein_ID = 0;
            Spielfeld = new Spielfeld(row, col, 0);
            Spielregeln = new Spielregeln(row, col, Spielfeld, win);
        
            MyWindow.setSpielernamen(Spieler[0].getSpielerName(), Spieler[1].getSpielerName());
            MyWindow.setSpielerFarbe(Spieler[0].getSpielerfarbe(), Spieler[1].getSpielerfarbe());
            MyWindow.setGoal(win);
        }

        public void neuesSpiel()
        {
            Spieleranzahl = 2;
            generate_Stein_ID = 0;
            Spielfeld = new Spielfeld(row, col, 0);
            Spielregeln = new Spielregeln(row, col,Spielfeld, win);
            MyWindow.setGoal(win);
        }

        public void pressedButton(int row, int col)
        {
            Farbe far = Spieler[AnDerReihe].getSpielerfarbe();
            int id = generate_Stein_ID;
            ISpielstein Stein = new Spielstein(far, id);
            if (Spielregeln.istGueltigerZug(row, col, Stein))
            {
                if (Spielregeln.istGewonnen(far))
                {
                    MyWindow.setButton(row, col, far);
                    gewonnenesSpiel(Spieler[AnDerReihe]);
                }
                else
                {
                    generate_Stein_ID++;
                    MyWindow.setButton(row, col, far);

                }

                AnDerReihe = (AnDerReihe + 1) % Spieleranzahl;
                MyWindow.setArrow(AnDerReihe);
            }
            else
            {
                //Do nothing
            }
        }

        private void gewonnenesSpiel(ISpieler Spieler)
        {
            Spieler.gewonnen();
            MyWindow.gewonnenesSpiel(Spieler.getSpielerName(), Spieler.getSpielerID(), Spieler.gewonneneSpiele());
        }


    }
}

File: TTT Python PySimpleGui/controller/Controller_Spiel.py
from controller.IController import IController
from model.Spieler import Spieler
from model.Spielfeld import Spielfeld
from model.Spielregeln import Spielregeln
from model.Farbe import Farbe
from model.Spielstein import Spielstein
from view.gui import gui


class Controller_Spiel(IController):
    def __init__(self, rows: int = 3, cols: int = 3, wins: int = 3):
        self._rows = rows
        self._cols = cols
        self._wins = wins
        self._spieleranzahl = 2
        spieler1 = Spieler(Farbe.Azure, "Alice 'O' Little", 0)
        spieler2 = Spieler(Farbe.Green, "Cassandra 'X' Cain", 1)
        self._DieSpieler = [spieler1, spieler2]
        self._AnDerReihe: int = 0
        self._generiere_Stein_ID = 0
        self._Feld = Spielfeld(rows, cols, 0)
        self._Regeln = Spielregeln(self._Feld, wins)
        self._gui = gui(self)
        self._gui.setSpieler(spieler1.name, spieler2.name, spieler1.Farbe, spieler2.Farbe)
        self._gui.start_up()

    def neuesSpiel(self) -> None:
        self._Feld = Spielfeld(self._rows, self._cols, 0)
        self._Regeln = Spielregeln(self._Feld, self._wins)

    def Button_pressed(self, row: int = -1, col: int = -1) -> None:
        far: Farbe = self._DieSpieler[self._AnDerReihe].Farbe
        stein_id = self._generiere_Stein_ID
        Stein = Spielstein(far, stein_id)
        if self._Regeln.ist_gueltiger_Zug(Stein, row, col):

            if self._Regeln.istGewonnen(far):
                self._gui.update_Button(row, col, far)
                self._gewonnen(self._DieSpieler[self._AnDerReihe])
            else:
                self._generiere_Stein_ID += 1
                self._gui.update_Button(row, col, far)

            self._AnDerReihe = (self._AnDerReihe + 1) % self._spieleranzahl
            self._gui.setArrow(self._DieSpieler[self._AnDerReihe].Farbe, self._AnDerReihe)

    def _gewonnen(self, sp: Spieler) -> None:
        sp.inc_wins()
        self._gui.Game_won(sp.wins, sp.id, sp.name)
